node/cli/src/command.rs

```diff
@@ -387,68 +383,96 @@ pub fn run() -> Result<()> {
 			})
 		},
 		None => {
-			let runner = cli.create_runner(&cli.run.normalize())?;
-
-			runner.run_node_until_exit(|config| async move {
-				let para_id = chain_spec::Extensions::try_get(&*config.chain_spec)
-					.map(|e| e.para_id)
-					.ok_or("Could not find parachain ID in chain-spec.")?;
-
-				#[cfg(feature = "frequency-rococo-local")]
-				if cli.instant_sealing {
-					return frequency_service::service::frequency_dev_instant_sealing(config, true)
-						.map_err(Into::into)
-				} else if cli.manual_sealing {
-					return frequency_service::service::frequency_dev_instant_sealing(config, false)
-						.map_err(Into::into)
-				}
-
-				let id = ParaId::from(para_id);
+			#[cfg(feature = "frequency-rococo-local")]
+			return run_local(cli);
+
+			#[cfg(any(
+				not(feature = "frequency-rococo-local"),
+				feature = "all-frequency-features"
+			))]
+			return run_parachain(cli);
+		},
+	}
+}
 
-				let parachain_account =
-					AccountIdConversion::<polkadot_primitives::v2::AccountId>::into_account_truncating(&id);
+#[cfg(feature = "frequency-rococo-local")]
+fn run_local(cli: Cli) -> Result<()> {
+	let runner = cli.create_runner(&cli.run.normalize())?;
+	runner.run_node_until_exit(|config| async move {
+		if cli.instant_sealing {
+			return frequency_service::service::frequency_dev_instant_sealing(config, true)
+				.map_err(Into::into)
+		} else {
+			return frequency_service::service::frequency_dev_instant_sealing(config, false)
+				.map_err(Into::into)
+		}
+	})
+}
 
-				let state_version = Cli::native_runtime_version(&config.chain_spec).state_version();
-				let block: Block = generate_genesis_block(&*config.chain_spec, state_version)
-					.map_err(|e| format!("{:?}", e))?;
-				let genesis_state = format!("0x{:?}", HexDisplay::from(&block.header().encode()));
+#[cfg(any(not(feature = "frequency-rococo-local"), feature = "all-frequency-features"))]
+use codec::Encode;
```

Comment:
I've found for these shorter ones it is usually better to either go fully qualified or to put the use statement inside the function. However not always the best choice. Depends a lot on the setup.

Reply:
That was existing code.  I can change it if you think it's necessary.